### DISCUSSION TITLE
Using `pandas` in a PR I'm planning

### DISCUSSION THREAD

**Discussion by Silloky** [tskit-dev/tskit#2973] [General]
Hi.
I'm currently working on implementing a reciprocal monophyly test for the `Tree` class.
My original code works with `pandas`, but I've noticed `pandas` isn't imported/used in `trees.py` script.
Can I still use it with my PR, or is it best to stick to complicated dictionaries ?
Thanks !

**Comment by jeromekelleher** [comment]
Hi @Silloky, welcome :wave:

Thanks for asking the question - you're right that we don't want to add an runtime dependency on pandas, as we're trying to keep the tskit core as lightweight as we can. However, testing the implementation using pandas would be totally fine.

So: how about you implement the version you have in mind using pandas, and open a PR so we can discuss how to proceed from there?

Thanks for contributing to tskit!

**Comment by Silloky** [comment]
@jeromekelleher Thanks for your answer !
Alright, I'll fork it and implement it with pandas for the moment, and then we can discuss a non-pandas solution.

By the way, there are probably many technical reasons for not using pandas for the tables, but I think it would be so much easier.
I've had to work on a project and within the first lines of code, I finally converted the table to a pandas dataframe because I found it easier to manipulate.
I understand it's very difficult (and somewhat useless - if it works, don't fix it) to switch to pandas dataframes, but from my point of view, it would be a great long-term improvement.

Anyway, I'll code the monophyly test and submit a PR.
Have a nice day !

**Reply by jeromekelleher** [reply]
It's good point @Silloky, and for Python-only use then pandas would probably provide a lot of benefits. But, tskit is a cross-language project with support for C and Rust, so switching over to pandas wouldn't help at all there and leave us with a lot of duplicated code etc.

Thanks a lot!

**Reply by Silloky** [reply]
Ah... OK, I forgot about the cross-compat with C.
I understand now. Thanks for developing this great lib (although sometimes it is frustrating 😁).
I do not know C, so will someone take care of adding my function to C as well ?

**Reply by jeromekelleher** [reply]
Let's see what it looks like, we might be able to do things with just numpy.

**Comment by Silloky** [comment]
OK, closing this discussion now @jeromekelleher as I have published the PR.
